pkg/apis/servicecatalog/serialization_test.go

```diff
@@ -0,0 +1,306 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicecatalog_test
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/rand"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	proto "github.com/golang/protobuf/proto"
+	flag "github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/testapi"
+	apitesting "k8s.io/kubernetes/pkg/api/testing"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	metav1 "k8s.io/kubernetes/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/runtime/schema"
+	"k8s.io/kubernetes/pkg/util/diff"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+
+	_ "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/install"
+)
+
+func init() {
+	testapi.Groups[servicecatalog.GroupName] = serviceCatalogAPIGroup()
+}
+
+// BABYNETES: ripped from pkg/api/serialization_test.go
```

Comment:
I think it would probably be worth, for new people coming onto the project, feeding back some of the commenting changes I made in kubernetes/kubernetes#39915 in this test.